### ISSUE TITLE
`triple-single-quotes` (`D300`) - false positive when docstring ends with a double quote

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#13838]
if i have a docstring that ends with a double quote, it's a syntax error:

```py
def foo():
    """this function does "a thing"""" # syntax error
```

so i decided to use single quotes instead, but this triggers D300:

```py
def foo():
    '''this function does "a thing"''' # error: Use triple double quotes `"""`
```

an alternative workaround would be to split the docstring onto multiple lines, but that triggers D200:

```py
def bar():
    # error: One-line docstring should fit on one line
    """
    this function does "a thing"
    """
```

**Comment by charliermarsh** [comment]
I think this has come up before, and I personally consider it correct to still flag `D300` here.

**Comment by DetachHead** [comment]
what would you recommend in cases like this where the docstring needs to end with a double quote?

**Comment by MichaReiser** [comment]
I'm leaning towards making this change because it increases formatter compatibility. 

The formatter accepts the single quotes in this specific use case. An alternative form of writing the above is to escape the quote:

```python
def foo():
    """this function does "a thing\"""" # syntax error
```

**Comment by TheRealBecks** [comment]
At least that one is wrong in general:
```
def foo():
    """this function does "a thing""""
```
Should be:
```
def foo():
    """this function does "a thing".""" # syntax error
```
->  Adding a `.` does the trick and the validation is working again, so you will get several other `D` errors (that's a good thing!). `The first line should end with a period` is rule `D400`.

In general a multiline docstring will be like:
```
"""
Replace a value.

Here comes more text for "a thing".
"""
```

So it looks like `ruff` is always failing due to `""""` (4x `"`) instead of `"""`. If that one gets fixed the parsing will work again for singleline and multiline comments.

**Comment by TheRealBecks** [comment]
...I had to think about it again: Maybe it's not an issue as the Pylance linter in VS Code is also complaining with `String literal is unterminated`. So I tested it with your example **without** the `.`:

```
import inspect


def docstring_test():
    """this function does "a thing""""


print(docstring_test.__doc__)
```

```
$ uv run temp.py 
  File "/..../temp.py", line 5
    """this function does "a thing""""
                                     ^
SyntaxError: unterminated string literal (detected at line 5)
```
-> Python exception

...and **with** `.`:
```
import inspect


def docstring_test():
    """this function does "a thing"."""


print(docstring_test.__doc__)
```

```
$ uv run temp.py 
this function does "a thing".
```
-> works

So we're talking about a Python systax error here (I wasn't aware of that when I wrote my first comment).

The question is: Can the ruff linter provide a solution or is it not working as the language server is throwing the syntax error?

**Comment by DetachHead** [comment]
yeah the reason i wanted to use single quotes was to avoid the syntax error. but your previous comment about ending the line with a `.` sounds like the best solution. i just happened to have the rule that enforces it disabled. closing this issue
